functions/src/definitions/userHandlers.ts

```diff
@@ -792,6 +828,33 @@ async function onTextListReceipt(messageObj: Message, platform = "whatsapp") {
   return Promise.resolve(step)
 }
 
+async function addInstanceToDb(
+  id: string,
+  hasMatch: boolean,
+  messageRef: FirebaseFirestore.DocumentReference | null,
+  messageUpdateObj: Object | null = null,
+  instanceRef: FirebaseFirestore.DocumentReference,
+  instanceUpdateObj: Object
+) {
+  const messageIdRef = db.collection("messageIds").doc(id)
+  try {
+    await db.runTransaction(async (t) => {
+      const doc = await t.get(messageIdRef)
+      if (doc.exists) {
+        return
+      }
+      if (!hasMatch && !!messageRef && !!messageUpdateObj) {
```

Comment:
Just a note. For `if` conditions, you don't have to use `!!` actually. If assigning boolean values, then the `!!` is necessary.

Reply:
Noted!